[bnxt] Use updated DMA APIs

jw14812 opened the pull request:
Replace malloc_phys with dma_alloc, free_phys with dma_free, alloc_iob with alloc_rx_iob, free_iob with free_rx_iob, virt_to_bus with dma or iob_dma.

mcb30 commented on src/drivers/net/bnxt/bnxt.c:
```diff
@@ -1823,13 +1841,13 @@ static int bnxt_hwrm_ring_alloc ( struct bnxt *bp, u8 type )
 		req->length 	= ( u32 )bp->nq.ring_cnt;
 		req->logical_id = 0xFFFF; // Required value for Thor FW?
-		req->page_tbl_addr = virt_to_bus ( bp->nq.bd_virt );
+		req->page_tbl_addr = ( unsigned long ) ( bp->nq.bd_virt );
```
This looks implausible to me.  The old code would have filled `page_tbl_addr` with a physical (bus) address, but the replacement code looks as though it fills `page_tbl_addr` with a virtual address.

jw14812 replied:
Ack, and fixed

mcb30 commented on src/drivers/net/bnxt/bnxt.c:
```diff
@@ -2116,8 +2134,8 @@ int bnxt_hwrm_run ( hwrm_func_t cmds[], struct bnxt *bp )
 	for ( ptr = cmds; *ptr; ++ptr ) {
-		memset ( bp->hwrm_addr_req,  0, REQ_BUFFER_SIZE );
-		memset ( bp->hwrm_addr_resp, 0, RESP_BUFFER_SIZE );
+		memset ( ( void * )bp->req_addr_mapping,  0, REQ_BUFFER_SIZE );
```
Doesn't necessarily need to be changed, but an explicit cast to `void *` is not necessary.

mcb30 commented on src/drivers/net/bnxt/bnxt.c:
```diff
@@ -2152,9 +2170,6 @@ static void bnxt_tx_adjust_pkt ( struct bnxt *bp, struct io_buffer *iob )
 	if ( iob_len ( iob ) != prev_len )
 		prev_len = iob_len ( iob );
-
-	iob_pad ( iob, ETH_ZLEN );
```
The removal of TX padding seems to be unrelated to using the DMA APIs.  Is this change intentional?

jw14812 replied:
Restored this change, and will remove the unnecessary padding in a future commit

mcb30 commented on src/drivers/net/bnxt/bnxt.h:
```diff
@@ -933,6 +933,19 @@ struct bnxt {
+	struct dma_mapping        rx_mapping;
+	struct dma_mapping        cq_mapping;
+	struct dma_mapping        nq_mapping;
+	dma_addr_t                tx_addr_mapping;
```
There's no real need for these `dma_addr_t` fields, or for having `dma_addr_t` defined as a type specific to the Broadcom driver.  Since you're cleaning up the DMA code anyway, I would suggest deleting the `dma_addr_t` typedef in `bnxt.h`, removing all of these fields within `struct bnxt`, and instead using `dma()` at the point of use for each of these values.

jw14812 replied:
Ack, removed this and created macros to do the dma() calls for better readability

jw14812 commented:
@mcb30 - Hi Michael, I have pushed some changes to address your comments.  When convenient can you check if they have addressed your concerns.  Thanks

mcb30 commented:
Looks plausible to me, thank you for updating.  Will QA and merge shortly.

mcb30 commented:
All merged, thank you!
